federation/schema.go

```diff
@@ -83,6 +92,53 @@ func convertVersionedSchemas(schemas serviceSchemas) (*SchemaWithFederationInfo,
 	fieldInfos := make(map[*graphql.Field]*FieldInfo)
 	for _, service := range serviceNames {
 		for _, typ := range serviceSchemasByName[service].Schema.Types {
+
+			// For federated fields parse the arguments to figure out which
+			// fields are the federated keys. They annotate that information
+			// on the field object.
+			if typ.Name == "Federation" {
+				for _, field := range typ.Fields {
+					for _, arg := range field.Args {
+
+						// Extract the type name from the formatting <object>-<service>
+						rootType := getRootType(arg.Type)
+						names := strings.SplitN(field.Name, "-", 2)
+						if len(names) != 2 {
+							return nil, oops.Errorf("Field %s doesnt have an object name and service name", field.Name)
+						}
+						objName := names[0]
+						inputType, ok := types[rootType.Name].(*graphql.InputObject)
+						if !ok {
+							return nil, oops.Errorf("Object %s is not an input object, but it is an argument to the field %s", rootType.Name, field.Name)
+						}
+
+						// For the object type on the current service
+						for _, t := range serviceSchemasByName[service].Schema.Types {
+							if objName == t.Name {
```

Comment:
Do you think we can remove this loop and the check?

Reply:
yup! removed!